src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -106,11 +106,31 @@ const clientRenderOnServerString = async (element, errorCount = 0) => {
   return clientElement;
 };
 
-const clientRenderOnBadMarkup = (element, errorCount = 0) => {
+function BadMarkupExpected() {}
+
+const clientRenderOnBadMarkup = async (element, errorCount = 0) => {
+  // First we render the top of bad mark up.
   var domElement = document.createElement('div');
   domElement.innerHTML =
     '<div id="badIdWhichWillCauseMismatch" data-reactroot="" data-reactid="1"></div>';
```

Comment:
It turns out I can't put junk text in there because `<div dangerouslySetInnerHTML={...} />` won't actually hit the patch up path so it fails on bad markup. However, I think we're fine not patching up the tree in that case since comparing would be difficult and expensive.